src/stores/index.js

```diff
@@ -851,23 +896,38 @@ const Store = Reflux.createStore({
     // Set the connection's app name to the electron app name of Compass.
     connection.appname = electron.remote.app.getName();
 
-    const dataService = new DataService(connection);
+    if (!this.state.isConnecting) {
+      // The connection attempt might have been cancelled while
+      // we were parsing the connection information, so we return here.
+      return;
+    }
 
     try {
-      const runConnect = promisify(dataService.connect.bind(dataService));
-      const connectedDataService = await runConnect();
+      const dataService = new DataService(connection);
+      this.connectingConnectionAttempt = createConnectionAttempt(dataService);
```

Comment:
what about `currentConnectionAttempt`?

Reply:
that good lol I was thinking it sounded funny connecting connection